posthog/models/property.py

```diff
@@ -62,35 +62,40 @@ def _parse_value(self, value: ValueT) -> Any:
         except (json.JSONDecodeError, TypeError):
             return value
 
-    def property_to_Q(self) -> Q:
+    def property_to_Q(self, table: Optional[TableWithProperties]) -> Q:
         from .cohort import CohortPeople
 
         value = self._parse_value(self.value)
+        if self.key == "distinct_id" and table == "events":
```

Comment:
I had added this check to the EE util functions but not the django ones which caused the issue because distinct_id was subbed in for person queries and the person table doesn't have a distinct_id column like the event table does.

Need tests here